### ISSUE TITLE
Use actionCartUpdateQuantityBefore return value

### ISSUE THREAD

**Issue by tom-combet** [PrestaShop/PrestaShop#27041]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

I'm developing a module and I need to abort cart quantity update if some constraint are violated. Furthermore, I'd like that the user get an error feedback in the front-office.
Currently, `actionCartUpdateQuantityBefore` execution return value is not used.
So I have to either :
- use `die` function but no user feedback
- use `actionFrontControllerInitAfter` but this is a very top-level/general hook so it's quite dirty and I don't benefit from the specific hook params

### Describe the solution you'd like

I'd like to return an array of errors or at least fill `$this->context->controller->errors[]` and return false in order to cancel the execution and display errors to the user.
I can do the PR but I'd like to have your approval before writing it. By the way, is it the only 'Before' hooks concerned by this issue?

**Comment by marwachelly** [comment]
Hello @Kaudaj 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments.
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider hiring a specialist to help you create it.

If you are willing to implement this feature into PrestaShop as a Pull Request, the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!
Thanks,

**Comment by kpodemski** [comment]
This hook is now available in the core:
https://devdocs.prestashop-project.org/8/modules/concepts/hooks/list-of-hooks/actioncartupdatequantitybefore/#information

**Comment by tom-combet** [comment]
I know, but that's not the issue: the hook execution still doesn't expect a return value.

> I'd like to return an array of errors or at least fill $this->context->controller->errors[] and return false in order to cancel the execution and display errors to the user.

**Comment by manudeluca** [comment]
Hi @tom-combet I have the same problem as you. Have you found a solution?

**Comment by tom-combet** [comment]
As I described in the issue description, I use the hook `actionFrontControllerInitAfter`.

**Comment by tom-combet** [comment]
Again, I don't know why this issue is closed.
It could be nice to answer why you close issues and PR for no reason guys...

@kpodemski Could you read my answer and explain me why it should remain closed if it's the case?

**Comment by kpodemski** [comment]
@tom-combet 

If I understand well, what you wanted is a different behavior of the hook or a way for it to have an impact for the validation. I think what happened was a misunderstanding (on my side).

I can reopen this issue if you are motivated to implement a way for this hook to meet your requirements.

btw. I'm not sure to which PR that were closed you are referring to?

**Comment by kpodemski** [comment]
I checked again and we do need to change the implementation here:
`Hook::exec('actionCartUpdateQuantityBefore', $data);`

To keep it BC break free, we could add an extra variable to `$data` and pass it by reference which would be something like:
`&$blockUpdateCart = false` 🤔 

what do you think?

**Comment by tom-combet** [comment]
> btw. I'm not sure to which PR that were closed you are referring to?

Nah, it was about an other PR not related to this issue, but whatever.
Thanks for answering, I appreciate! It's not cool when you feel like you're talking to a wall 😅

**Comment by kpodemski** [comment]
@tom-combet It's hard to keep up with all the updates on GitHub sometimes. We're only a few people there. Remember that you can always try to reach out to us on Slack 👍🏻

**Comment by tom-combet** [comment]
I forgot to answer to initial issue tho 😆 

Is it a BC break if we use the hook return value, like this?

```php 
if (Hook::exec('actionCartUpdateQuantityBefore', $data) === false) {
    return false;
}
```
